### PR TITLE
profiling: support auto collection

### PR DESCRIPTION
**Pull request by rleungx**
This PR supports auto profiling.

### CODE REVIEW

File: pkg/apiserver/profiling/service.go [hunk 1]
```diff
@@ -131,8 +168,79 @@ func (s *Service) start(c *gin.Context) {
...
+// @Summary Start auto profiling
+// @Description Start auto profiling
+// @Produce json
+// @Param req body StartRequest true "auto profiling request"
+// @Security JwtAuth
+// @Success 200 {string} string "success"
+// @Failure 400 {object} utils.APIError
+// @Router /profiling/auto/start [post]
+func (s *Service) autoStart(c *gin.Context) {
+	if atomic.LoadUint32(&s.isAutoCollectRunning) > 0 {
```
Comment (HunDunDM) [inline pkg/apiserver/profiling/service.go]: Would it be better to use `CompareAndSwap`?

Comment (breeswish) [reply pkg/apiserver/profiling/service.go]: Yes, we need to use CAS.

File: pkg/apiserver/profiling/service.go [hunk 2]
```diff
@@ -131,8 +168,79 @@ func (s *Service) start(c *gin.Context) {
...
+}
+
+// @Summary Stop auto profiling
+// @Description Stop auto profiling
+// @Produce json
+// @Security JwtAuth
+// @Success 200 {string} string "success"
+// @Router /profiling/auto/stop [post]
+func (s *Service) autoStop(c *gin.Context) {
+	if atomic.LoadUint32(&s.isAutoCollectRunning) == 0 {
```
Comment (breeswish) [inline pkg/apiserver/profiling/service.go]: CAS

File: pkg/apiserver/profiling/service.go [hunk 3]
```diff
@@ -131,8 +168,79 @@ func (s *Service) start(c *gin.Context) {
...
+		c.JSON(http.StatusBadRequest, "auto profiling job has been stopped")
+	}
+
+	atomic.StoreUint32(&s.isAutoCollectRunning, 0)
+	c.JSON(http.StatusOK, "success")
```
Comment (breeswish) [inline pkg/apiserver/profiling/service.go]: we can use APIEmptyResponse

**Comment by rleungx** [comment]
PTAL @breeswish @HunDunDM

File: pkg/apiserver/profiling/service.go [hunk 4]
```diff
@@ -131,8 +168,78 @@ func (s *Service) start(c *gin.Context) {
...
+		if atomic.LoadUint32(&s.isAutoCollectRunning) == 0 {
+			return
+		}
+		select {
+		case <-s.ctx.Done():
+			return
+		case <-timer.C:
+			_, _ = s.collect(req)
+			timer.Reset(interval)
+		default:
```
Comment (HunDunDM) [inline pkg/apiserver/profiling/service.go]: This `for` loop does not block.

**Comment by HunDunDM** [comment]
close by #439

**Pull request closed without merging**